### DISCUSSION TITLE
Navigation in local dependencies (maven project)

### DISCUSSION THREAD

**Discussion by gsavini-mce** [scalameta/metals#5685] [Q&A] [answered]
Is it possibile to navigate to local dependencies in a maven project?

I am able to go to the definitions of remote dependencies, but not to the local ones (jars and other scala projects).

Example:
- ProjectA installed with mvn install
- LibraryC (jar) installed with mvn install
- ProjectB depends on ProjectA and LibraryC
- ProjectA sees the symbols defined in ProjectA and LibraryC, but it's not possible to navigate to the definitions.

**Comment by tgodzik** [comment, marked as answer]
Thanks for the question. Metals currently relies on source jar, so ideally if you have separate library published locally you should also publish sources.

The alternative would be decompiling classfiles, which we could work on, but this gives a much worse experience for the user.
